app/services/notifiers/plage_ouverture_destroyed.rb

```diff
@@ -0,0 +1,14 @@
+class Notifiers::PlageOuvertureDestroyed < Notifiers::PlageOuvertureBase
+  def initialize(plage_ouverture, motif_ids)
```

Comment:
Je ne suis pas un grand fan des arguments positionnels en règle générale, ici ça serait pas mal de mettre un kwarg au moins sur motif_ids, qu’en penses tu ?

```suggestion
  def initialize(plage_ouverture, motif_ids:)
```

voire même partout même pour l’argument principal plage_ouverture

```suggestion
  def initialize(plage_ouverture:, motif_ids:)
```

Reply:
et sinon si tu as la motive de réfléchir à une manière plus élégante de refacto ce job de sérialisation on peut envisager qqch comme ça : 

```rb
class Notifiers::PlageOuvertureDestroyed
  def initialize(plage_ouverture)
    @plage_attributes_serialized = PlageOuverture.serialize_for_active_job(plage_ouverture).merge(motif_ids: plage_ouverture.motif_ids)
  end

  def notify
    Agents::PlageOuvertureMailer.with(plage_ouverture: @plage_attributes_serialized)
  end
end
```

puis dans l’appelant : 

```rb
    notifier = Notifiers::PlageOuvertureDestroyed.new(@plage_ouverture)
    if @plage_ouverture.destroy
      notifier.perform
      ...
```

Reply:
Effectivement, j'avais essayé de récupérer les motifs comme tu proposes, mais comme le destroy avait été appliqué, je n’avais pas les motif_ids

Mais c’est vrai qu’en remontant le new et en faisant le perform après ça devrait parfaitement fonctionner 

Je vais faire ça